### DISCUSSION TITLE
Home Assistant and Nabu casa URL is to long for Home Connect

### DISCUSSION THREAD

**Discussion by Hellis81** [alexkn/node-red-contrib-homeconnect#45] [General]
When I use this flow and copy the callback URI from the node the textbox in Home Connect can't take the full URI.

The URI is something like https://blabla.nabu.casa/api/hassio_ingress/blabla/homeconnect/auth/callback
The textbox can only hanle: https://blabla.nabu.casa/api/hassio_ingress/blabla/homeconnect 
The rest does not fit

I have reported the issue to Home Connect but figured you might want to know since it makes the node not work.

**Comment by Blackhawk386** [comment]
Hi. Same problem for me here, did you figure out any solution? Probably there are two ways of solution. One is the limitation of signs from Home Connect side. The second one could be a shortened URI coming from node itself. There are 43 signs between ...ingress/ and /homeconnect.... If it is possible to shorten this by 14 signs, the node should work,

**Reply by Hellis81** [reply]
In my opinion the fault is at Bosch.
A URL could be far longer than what they limit the text box to.
Up to 2000 characters shouldn't be an issue these days. 
Bosch is using non industry standard here and should change it. 

https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers

Feel free to send them and Email. I will do it again.

**Reply by Hellis81** [reply]
@Blackhawk386 did you send them an email? 
I have not heard back from them since I sent my last email.

**Reply by Blackhawk386** [reply]
@Hellis81 sorry, I thought I presented my workaround here. You can use your internal IP for authorization, so when you enter https://192.168.XXX.XXX:8123/api/hassio_ingress/XXXXXXX/homeconnect/auth/callback it is possible to enter the URI in the home connect portal and the authorization will work.

**Reply by Hellis81** [reply]
@Blackhawk386  That did not work for me. I get an error that the connection is not secure when I use https, and when I use http I get unahtorized.

**Reply by Hellis81** [reply]
After a few more attempts I get state or code missing... but then in the callback URL there is a code.

```
Missing code or state parameter in https://foobar.ui.nabu.casa/auth/external/callback?code=eyJ4LXJlZyI6IkVVIiwieC1lbnYiOiJQUkQiLCJ0b2tlbiI6IjAyZTM5ZmNkLTE1NzUtNDdiYi1iM2Q5LTU2NGE1ODVjYjQ5ZSIsImNsdHkiOiJwcml2YXRlIn0%3D&grant_type=authorization_code
```
